src/librustc/session/config.rs

```diff
@@ -1968,6 +1948,18 @@ pub fn build_session_options_and_crate_config(matches: &getopts::Matches)
 
     let crate_name = matches.opt_str("crate-name");
 
+    let remap_path_prefix = matches.opt_strs("remap-path-prefix")
+        .into_iter()
+        .map(|remap| {
+            let mut parts = remap.rsplitn(2, '=');
+            match (parts.next(), parts.next()) {
```

Comment:
The evaluation of components in tuple constructors is *probably* defined, but I still find this hard to read. And it's probably the wrong order because the iterator starts at the end `:)` Could you rewrite this to something like:
```rust
let mut parts = ...;
let to = parts.next();
let from = parts.next();
match (from, to) { ... }
```

Reply:
Is there anything that implements `FromIterator<Item=T>` for `Option<(T,...)>`? I keep ending up in situations where it would be useful (mostly parsing with split), but I suspect it would clash with an existing `FromIterator` for `Option<T>` (unless specialization helps?).